src/datanode/src/config.rs

```diff
@@ -53,6 +53,18 @@ pub enum ObjectStoreConfig {
     Gcs(GcsConfig),
 }
 
+impl ObjectStoreConfig {
+    pub fn extract_variant_name(&self) -> &'static str {
```

Comment:
Looks like `name` is better.
```suggestion
    pub fn name(&self) -> &'static str {
```